kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/apps/v1/RollingUpdater.java

```diff
@@ -222,25 +220,22 @@ private void waitUntilPodsAreReady(final T obj, final String namespace, final in
         }
       }
       podCount.set(count);
-      if (count == requiredPodCount) {
-        countDownLatch.countDown();
-      }
-    };
-
-    ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
-    ScheduledFuture poller = executor.scheduleWithFixedDelay(readyPodsPoller, 0, 1, TimeUnit.SECONDS);
-    ScheduledFuture logger = executor.scheduleWithFixedDelay(() -> LOG.debug("Only {}/{} pod(s) ready for {}: {} in namespace: {} seconds so waiting...",
-        podCount.get(), requiredPodCount, obj.getKind(), obj.getMetadata().getName(), namespace), 0, loggingIntervalMillis, TimeUnit.MILLISECONDS);
+      return count == requiredPodCount;
+    });
+
+    ScheduledFuture<?> logger = Utils.scheduleAtFixedRate(Utils.getCommonExecutorSerive(),
```

Comment:
` Th `Utils#scheduleAtFixedRate` is really misleading. We should rename it to match the delegated`CachedSingleThreadScheduler#scheduleWithFixedDelay` method name.

Reply:
It's a little confusing, but it actually is scheduleAtFixedRate.

The logic is handing the task off to another executor, so unless it's an executor that executes in the current thread, it means that fixed delay is actually a fixed rate.

Reply:
OK, makes sense, I missed the `executor.execute(...)` part of the method.

Reply:
Related to #3016 -> #3031